### PR TITLE
Mysql migration from internal MySQL databases to Azure Databases for MySQL

### PR DESCRIPTION
**Pull request by gossion**
### Changelog

* Add document and scripts to migrating internal MySQL databases to Azure Databases for MySQL

### CODE REVIEW

**Comment by gossion** [comment]
@bingosummer . No, BBR is not designed for migration, it is designed for disaster recovery, its backups are intended to be restored to an environment deployed with the same configuration.

File: docs/advanced/configure-cf-external-databases-using-azure-mysql-postgres-service/migrate-mysql-databases/README.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
+# Migrate Cloud Foundry MySQL databases to Azure Database for MySQL
+
+On Cloud Foundry (CF), it is recommended to use PAAS service for the purpose of high availability. [Azure Database for MySQL](https://azure.microsoft.com/en-us/services/mysql/) is one of services that CF can levege for HA. 
+
+For a new deployment, you can refer to this [guidance](../README.md) to use Azure Database for MySQL as external databases.
+For an existed deployment, if you want to use Azure Database for MySQL, this document will provide a simple guildance to migrate the database.
+
+**NOTE:** Be careful on operating databases, you should practise on a test environment before officially do the migration on your production environment.
```
Comment (andyliuliming) [inline docs/advanced/configure-cf-external-databases-using-azure-mysql-postgres-service/migrate-mysql-databases/README.md]: hmmm, think about removing this paragraph, look like a bit redundant.

File: docs/advanced/configure-cf-external-databases-using-azure-mysql-postgres-service/migrate-mysql-databases/README.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+In this guidance, we will use `mysqldump` and `mysql` command-lines to dump backup files from internal databases and then retore them to Azure Databases for MySQL.
+
+1. Figure out databases to be migrated
+
+    Check your CF manifest carefully, find out which databases are used. Write down database name, user, password for  step `Migrate databases`. If you are using Ops Manager, follow this [guid](https://docs.pivotal.io/pivotalcf/2-2/customizing/credentials.html) to get credentials.
```
Comment (andyliuliming) [inline docs/advanced/configure-cf-external-databases-using-azure-mysql-postgres-service/migrate-mysql-databases/README.md]: extra white space in "password for  step"

Comment (andyliuliming) [inline docs/advanced/configure-cf-external-databases-using-azure-mysql-postgres-service/migrate-mysql-databases/README.md]: follow this [guidance]?

File: docs/advanced/configure-cf-external-databases-using-azure-mysql-postgres-service/migrate-mysql-databases/README.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+1. Prepare Azure Database for MySQL
+
+    * Follow guidance on this [link](../README.md#prepare-azure-mysqlpostgres-database), please remember to turn on `Allow access to Azure services` and Disable `Enforce SSL connection in Azure MySQL`.
+    * Follow guidance on this [link](../README.md#create-the-databases) to create releated databases.
+
+1. Stop CC VMs
+
+    * Turn off bosh resurrector (`bosh -e azure -d cf  update-resurrection off`)
+    * On Azure portal, stop all the CC VM jobs (Cloud Controller, Cloud Controller Worker, and Clock Global). Note that your CF API services will stop being available at this point (running apps should continue to be available though)
```
Comment (andyliuliming) [inline docs/advanced/configure-cf-external-databases-using-azure-mysql-postgres-service/migrate-mysql-databases/README.md]: change from "stop all the CC VM jobs" to "stop all the vms which the cc jobs are running on"?

File: docs/advanced/configure-cf-external-databases-using-azure-mysql-postgres-service/migrate-mysql-databases/README.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    * `bosh cck` the cf deployment to check for any errors with the bosh state. It should ask you if you want to delete references to the missing CC jobs, which you want to do.
```
Comment (andyliuliming) [inline docs/advanced/configure-cf-external-databases-using-azure-mysql-postgres-service/migrate-mysql-databases/README.md]: what does it mean "which you want to do"?

File: docs/advanced/configure-cf-external-databases-using-azure-mysql-postgres-service/migrate-mysql-databases/README.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+1. Migrate databases
+
+    For each database, you need to dump the database and then restore it manually. For example, let us say the VM that hosts the databases is `10.0.0.7`, you can use the commands below to migrate `cloud_controller`:
+
+    ```
+    mysqldump -u cloud_controller -p cloud_controller -h 10.0.16.7 --single-transaction > cloud_controller.sql
```
Comment (andyliuliming) [inline docs/advanced/configure-cf-external-databases-using-azure-mysql-postgres-service/migrate-mysql-databases/README.md]: better to move the databases at the same time so it's more possible that the database in consistence?

File: docs/advanced/configure-cf-external-databases-using-azure-mysql-postgres-service/migrate-mysql-databases/README.md (hunk 4 above)
Comment (gossion) [reply docs/advanced/configure-cf-external-databases-using-azure-mysql-postgres-service/migrate-mysql-databases/README.md]: copied from https://github.com/pivotal-cf/goblob/blob/master/README.md#removing-nfs-post-migration. 
It means that you should delete references.

File: docs/advanced/configure-cf-external-databases-using-azure-mysql-postgres-service/migrate-mysql-databases/README.md (hunk 3 above)
Comment (gossion) [reply docs/advanced/configure-cf-external-databases-using-azure-mysql-postgres-service/migrate-mysql-databases/README.md]: copied from https://github.com/pivotal-cf/goblob/blob/master/README.md#removing-nfs-post-migration. But I think your wording is better, will change.

File: docs/advanced/configure-cf-external-databases-using-azure-mysql-postgres-service/migrate-mysql-databases/README.md (hunk 4 above)
Comment (andyliuliming) [reply docs/advanced/configure-cf-external-databases-using-azure-mysql-postgres-service/migrate-mysql-databases/README.md]: so our customer should choose yes for this?

File: docs/advanced/configure-cf-external-databases-using-azure-mysql-postgres-service/migrate-mysql-databases/README.md (hunk 1 above)
Comment (gossion) [reply docs/advanced/configure-cf-external-databases-using-azure-mysql-postgres-service/migrate-mysql-databases/README.md]: no harm to remind, will keep this paragraph. thanks.

File: docs/advanced/configure-cf-external-databases-using-azure-mysql-postgres-service/migrate-mysql-databases/README.md (hunk 2 above)
Comment (gossion) [reply docs/advanced/configure-cf-external-databases-using-azure-mysql-postgres-service/migrate-mysql-databases/README.md]: good catch, will change.

File: docs/advanced/configure-cf-external-databases-using-azure-mysql-postgres-service/migrate-mysql-databases/README.md (hunk 5 above)
Comment (gossion) [reply docs/advanced/configure-cf-external-databases-using-azure-mysql-postgres-service/migrate-mysql-databases/README.md]: every db has different credential(user,passwd). in most cases, they can't be dump with the same credential, so you can't move the DBs at the same time.

File: docs/advanced/configure-cf-external-databases-using-azure-mysql-postgres-service/migrate-mysql-databases/README.md (hunk 2 above)
Comment (gossion) [reply docs/advanced/configure-cf-external-databases-using-azure-mysql-postgres-service/migrate-mysql-databases/README.md]: will change

**Pull request merged**